### PR TITLE
Add validation to block usage of --ingress-host flag together with --…

### PR DESCRIPTION
**Pull request by danielm-codefresh**
…access-mode=tunnel

### CODE REVIEW

File: cmd/commands/runtime_install.go [hunk 1]
```diff
@@ -196,8 +195,11 @@ func NewRuntimeInstallCommand() *cobra.Command {
 
 			createAnalyticsReporter(ctx, reporter.InstallFlow, installationOpts.DisableTelemetry)
 
-			if (accessMode != "") {
+			if accessMode != "" {
 				installationOpts.AccessMode = platmodel.AccessMode(strings.ToUpper(accessMode))
+				if installationOpts.AccessMode == platmodel.AccessModeTunnel && installationOpts.IngressHost != "" {
+					return fmt.Errorf("ingress host can't be set when access mode is Tunnel.")
+				}
```
Comment (ghost) [inline cmd/commands/runtime_install.go]: maybe this should be after the next condition? make sure accessMode is valid first?

File: cmd/commands/runtime_install.go [hunk 2]
```diff
@@ -1145,7 +1146,11 @@ func preInstallationChecks(ctx context.Context, opts *RuntimeInstallOptions) (*r
 		return nil, err
 	}
 
-	rt := opts.DownloadRuntimeDef
+	runtimeDef := getRuntimeDef(opts.runtimeDef, opts.versionStr)
+	rt, err := runtime.Download(runtimeDef, opts.RuntimeName, opts.featuresToInstall)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download runtime definition: %w", err)
+	}
```
Comment (ghost) [inline cmd/commands/runtime_install.go:1154]: it doesn't make sense to keep the downloaded runtime definition on the opts? i think we can use it later. but if it is not actually being used, i guess there is no point in it

Comment (danielm-codefresh) [reply cmd/commands/runtime_install.go:1154]: I added this in the last version, where I moved runtime definition download to the pre command handler (where we can early check if CLI is compatible with runtime version), but now I noticed that we later check for features such as ingressless...

File: cmd/commands/runtime_install.go (hunk 1 above)
Comment (danielm-codefresh) [reply cmd/commands/runtime_install.go]: done

**Review by noam-codefresh — APPROVED** [review]

**Pull request merged**